packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go

```diff
@@ -344,11 +338,15 @@ func (dhtPeer *DHTPeer) handleDelegateService(ready *sync.WaitGroup) {
 
 	lerror, _, linfo, _ := dhtPeer.getLoggers()
 
+	done := false
 	for {
 		select {
 		default:
 			linfo().Msg("DelegateService listening for new connections...")
-			ready.Done()
+			if !done {
+				done = true
+				ready.Done()
```

Comment:
Notification mechanism to guarantee  that delegation service is running before `New` returns the newly created `DHTPeer`